api/src/main/java/org/apache/cloudstack/api/command/admin/storage/MigrateResourcesToAnotherSecondaryStorageCmd.java

```diff
@@ -0,0 +1,132 @@
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.cloudstack.api.command.admin.storage;
+
+import com.cloud.event.EventTypes;
+import org.apache.cloudstack.acl.RoleType;
+import org.apache.cloudstack.api.APICommand;
+import org.apache.cloudstack.api.ApiCommandResourceType;
+import org.apache.cloudstack.api.ApiConstants;
+import org.apache.cloudstack.api.BaseAsyncCmd;
+import org.apache.cloudstack.api.Parameter;
+import org.apache.cloudstack.api.response.ImageStoreResponse;
+import org.apache.cloudstack.api.response.MigrationResponse;
+import org.apache.cloudstack.api.response.SnapshotResponse;
+import org.apache.cloudstack.api.response.TemplateResponse;
+import org.apache.cloudstack.context.CallContext;
+
+import java.util.List;
+
+@APICommand(name = "migrateResourceToAnotherSecondaryStorage",
+            description = "migrates resources from one secondary storage to destination image store",
+            responseObject = MigrationResponse.class,
+            requestHasSensitiveInfo = false,
+            responseHasSensitiveInfo = false,
+            since = "4.19.0",
+            authorized = {RoleType.Admin})
+public class MigrateResourcesToAnotherSecondaryStorageCmd extends BaseAsyncCmd {
+
+
+    /////////////////////////////////////////////////////
+    //////////////// API parameters /////////////////////
+    /////////////////////////////////////////////////////
+
+    @Parameter(name = ApiConstants.SRC_POOL,
+               type = CommandType.UUID,
+               entityType = ImageStoreResponse.class,
+               description = "id of the image store from where the data is to be migrated",
+               required = true)
+    private Long id;
+
+
+    @Parameter(name = ApiConstants.DEST_POOL,
+               type = CommandType.UUID,
+               entityType = ImageStoreResponse.class,
+               description = "id of the destination secondary storage pool to which the resources are to be migrated",
+               required = true)
+    private Long destStoreId;
+
+    @Parameter(name = "templates",
+               type = CommandType.LIST,
+               collectionType = CommandType.UUID,
+               entityType = TemplateResponse.class,
+               description = "id(s) of the templates to be migrated",
+               required = false)
+    private List<Long> templateIdList;
+
+    @Parameter(name = "snapshots",
+               type = CommandType.LIST,
+               collectionType = CommandType.UUID,
+               entityType = SnapshotResponse.class,
+               description = "id(s) of the snapshots to be migrated",
+               required = false)
```

Comment:
are volumes intentionally not considered?

Reply:
Yes. As per my understanding, volumes don't exist on secondary storage by default. It's created in order to provide a user with a download link or for some intermediate action.

I couldn't find migration for volumes in existing implementation for secondary storage migration either.